code/modules/power/supermatter/supermatter.dm

```diff
@@ -352,36 +348,68 @@
 			else if(damage >= damage_archived) // The damage is still going up
 				radio.talk_into(src, "[warning_alert] Instability: [stability]%", engineering_channel, get_spans(), get_default_language())
 				lastwarning = REALTIMEOFDAY - (WARNING_DELAY * 5)
-
 			else                                                 // Phew, we're safe
 				radio.talk_into(src, "[safe_alert] Instability: [stability]%", engineering_channel, get_spans(), get_default_language())
 				lastwarning = REALTIMEOFDAY
-
 			if(power > POWER_PENALTY_THRESHOLD)
 				radio.talk_into(src, "Warning: Hyperstructure has reached dangerous power level.", engineering_channel, get_spans(), get_default_language())
 				if(powerloss_inhibitor < 0.5)
 					radio.talk_into(src, "DANGER: CHARGE INERTIA CHAIN REACTION IN PROGRESS.", engineering_channel, get_spans(), get_default_language())
-
 			if(combined_gas > MOLE_PENALTY_THRESHOLD)
 				radio.talk_into(src, "Warning: Critical coolant mass reached.", engineering_channel, get_spans(), get_default_language())
 
-		if(damage > explosion_point)
-			for(var/mob in GLOB.living_mob_list)
-				var/mob/living/L = mob
-				if(istype(L) && L.z == z)
-					if(ishuman(mob))
-						//Hilariously enough, running into a closet should make you get hit the hardest.
-						var/mob/living/carbon/human/H = mob
-						H.hallucination += max(50, min(300, DETONATION_HALLUCINATION * sqrt(1 / (get_dist(mob, src) + 1)) ) )
-					var/rads = DETONATION_RADS * sqrt( 1 / (get_dist(L, src) + 1) )
-					L.rad_act(rads)
-
-			explode()
-
+/obj/machinery/power/supermatter_shard/proc/SM_autoprocess_proximity()
+	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
+		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
+			var/D = sqrt(1 / max(1, get_dist(l, src)))
+			l.hallucination += (power * config_hallucination_power * D) * PROCESSING_HANDICAP
+			l.hallucination = Clamp(0, 200, l.hallucination)
+	for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
+		var/rads = ((power / 10) * sqrt( 1 / max(get_dist(l, src),1))) * PROCESSING_HANDICAP
+		l.rad_act(rads)
+	for(var/obj/machinery/power/rad_collector/R in GLOB.rad_collectors)
```

Comment:
Purely a balance suggestion: Add a check for has_gas and only run this energy pulse if there is gas. Otherwise players can cheese the supermatter by simply RCDing all floors away and getting around same amount of power as a safe mix.